src/App.tsx

```diff
@@ -1,14 +1,58 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useEffect,
+  useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { SelectValues } from './types/SelectValues';
+import { TodoModal } from './components/TodoModal';
 
 export const App: React.FC = () => {
+  const [isTodosLoaded, setIsTodosLoaded] = useState(false);
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [select, setSelect] = useState(SelectValues.all);
+  const [query, setQuery] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  function getPreparedTodos(todos: Todo[]): Todo[] {
+    let filtedTodos: Todo[] = [...todos];
+
+    switch (select) {
+      case SelectValues.active:
+        filtedTodos = filtedTodos.filter(todo => todo.completed === false);
+        break;
+
+      case SelectValues.completed:
+        filtedTodos = filtedTodos.filter(todo => todo.completed === true);
+        break;
+
+      default: filtedTodos = [...filtedTodos];
```

Comment:
```suggestion
      default:
        break;
```
anyway, you copied the array initially in the function, moreover, you will filter it below. So doing 1 additional copy doesn't make any sense